From 29ebd12dccf706414063eae9d6432efe41a2ca98 Mon Sep 17 00:00:00 2001
From: Alexander Martinz <amartinz@shiftphones.com>
Date: Wed, 4 Jul 2018 18:43:04 +0200
Subject: [PATCH 13/32] logging: prevent NPE at logDeepShortcutsOpen

getLaunchProviderRecursive(View) can return null.

Check if provider is null before using it.

Change-Id: I8cd157ba315e1dcc1fed428e0474b55b774bbd2f
Signed-off-by: Alexander Martinz <amartinz@shiftphones.com>
---
 src/com/android/launcher3/logging/UserEventDispatcher.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/com/android/launcher3/logging/UserEventDispatcher.java b/src/com/android/launcher3/logging/UserEventDispatcher.java
index d1e1051ee..2295cd9bd 100644
--- a/src/com/android/launcher3/logging/UserEventDispatcher.java
+++ b/src/com/android/launcher3/logging/UserEventDispatcher.java
@@ -367,7 +367,7 @@ public class UserEventDispatcher {
 
     public void logDeepShortcutsOpen(View icon) {
         LogContainerProvider provider = getLaunchProviderRecursive(icon);
-        if (icon == null || !(icon.getTag() instanceof ItemInfo)) {
+        if (icon == null || !(icon.getTag() instanceof ItemInfo) || provider == null) {
             return;
         }
         ItemInfo info = (ItemInfo) icon.getTag();
-- 
2.17.1

